pkgs/applications/emulators/dosbox-x/default.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, alsa-lib
+, autoconf
+, automake
+, file
+, fluidsynth
+, freetype
+, glib
+, libpcap
+, libpng
+, libslirp
+, libXext
+, libxkbfile
+, libXrandr
+, nasm
+, ncurses
+, pkg-config
+, SDL2
+, which
+, yad
+, zlib
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dosbox-x";
+  version = "2023.09.01";
+
+  src = fetchFromGitHub {
+    owner = "joncampbell123";
+    repo = "dosbox-x";
+    rev = "dosbox-x-v${finalAttrs.version}";
+    hash = "sha256-gEqIPXrq16p6VC9aGW2sRSWE0SG9hiMWeRzSoDvUbB0=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    file
+    pkg-config
+  ];
+
+  buildInputs = [
+    alsa-lib
+    fluidsynth
+    freetype
+    glib
+    libpcap
+    libpng
+    libslirp
+    libXext
+    libxkbfile
+    libXrandr
+    nasm
+    ncurses
+    SDL2
+    which
+    yad
+    zlib
+  ];
+
+  postPatch = ''
+    substituteInPlace vs/sdlnet/ltmain.sh vs/sdlnet/acinclude/libtool.m4 --replace /usr/bin/file ${file}/bin/file
+    patchShebangs .
+  '';
+
+  # dosbox-x really wants to use its own SDL_NET
+  # https://github.com/joncampbell123/dosbox-x/blob/5eea489cda7c1e6fbd1ded6932dd8459b5e0693a/build-sdl2#L42
+  preConfigure = ''
+    ./autogen.sh
+    pushd vs/sdlnet
+    ./build-dosbox.sh
+    popd
+  '';
+
+  configureFlags = [ "--enable-sdl2" ];
+  hardeningDisable = [ "format" ];
```

Comment:
Could you document the exact issue that format hardening causes? A link to an upstream issue would also be great - if upstream isn't aware of this yet then it should also be reported to them.

Reply:
I'll add a comment to https://github.com/joncampbell123/dosbox-x/issues/4436